recipes/diligent-tools/all/conanfile.py

```diff
@@ -58,7 +58,11 @@ def validate(self):
             raise ConanInvalidConfiguration("Can't build diligent tools as shared lib")
 
     def requirements(self):
-        self.requires("diligent-core/2.5.1")
+        if self.version == "2.5.2":
+            self.requires("diligent-core/2.5.2")
+            self.requires('taywee-args/6.3.0')
+        else:
+            self.requires("diligent-core/2.5.1")
```

Comment:
I don't think this is written with future diligent-tools releases in mind.
I would do it as follows:
```python
        self.requires(f"diligent-core/{self.version}")  # Require the same diligent-core version
        if self.version >= "2.5.2":
            self.requires('taywee-args/6.3.0')
```

or

```python
        diligent_core_version = {
            "2.5.1": "2.5.1",
            "2.5.2": "2.5.2",
        }.get(str(self.version), "2.5.2")  # require diligent-core/2.5.2 by default
        self.requires(f"diligent-core/{diligent_core_version}")
        if self.version >= "2.5.2":
            self.requires('taywee-args/6.3.0')
```

Reply:
done.
But there is no diligent-tools/2.5.1 tag, no tags were created back then